### PR TITLE
Improve navigation parameters reference

### PR DESCRIPTION
**Pull request by alexandra-simeonova**
**Description**

Changes proposed in this pull request:
- include the navigation example in a separate file
- add "type" & "required" sections to each parameter so developers can easily access this information
- change the format of the file with subheadings to make it easier to read (table form was also tested but did not work well in the case of "sub parameters/properties") 

**Related issue(s)**
Epic #766

### CODE REVIEW

**Comment by JohannesDoberer** [comment]
There are some question marks if a property is *required* or not. Also *type* is sometimes not clear. One developer should replace these question marks with the correct values.

File: docs/navigation-parameters-reference.md [hunk 1]
```diff
@@ -1,269 +1,293 @@
...
+You can configure the way Luigi tackles routing in your application in the `routing:` section of the configuration file. For example, you can choose the routing strategy to apply in your application as either hash or path location routing.
+
+### useHashRouting
+- **type**: boolean 
+- **required**: yes 
+- **description**: defines either hash-based (`url.com/#/yourpath`) or path-based (`url.com/yourpath`) routing.  
+
+### nodeParamPrefix
+- **type**: ? string 
+- **required**: ?
```
Comment (zarkosimic) [inline docs/navigation-parameters-reference.md]: - **type**: string 
- **required**: no

File: docs/navigation-parameters-reference.md [hunk 2]
```diff
@@ -1,269 +1,293 @@
...
+- **description**: sets the prefix character when using the `LuigiClient.linkManager().withParam()` function, which provides a way to simply attach query properties to the view URL for activities such as sorting and filtering. The URL contains the properties to allow deep linking. If you want to use a different character prefix, define yours here. The default character is `~`.
+
+### skipRoutingForUrlPatterns
+- **type**: string
+- **required**: ?
```
Comment (zarkosimic) [inline docs/navigation-parameters-reference.md]: - **type**: RegExp[]
- **required**: no

File: docs/navigation-parameters-reference.md [hunk 3]
```diff
@@ -1,269 +1,293 @@
...
+- **description**: defines regex patterns to be skipped by the router when listening for path changes. This property is used for excluding **redirect_uri** properties. Default patterns are `[/access_token=/, '/id_token=/]`.
+
+### pageNotFoundHandler
+- **type**: function  
+- **required**: ?
```
Comment (zarkosimic) [inline docs/navigation-parameters-reference.md]: - **type**: any  
- **required**: no

File: docs/navigation-parameters-reference.md [hunk 4]
```diff
@@ -1,269 +1,293 @@
...
+  - **wrongPath** (string): the path that user tried navigating to
+  - **wasAnyPathFitted** (bool): it is true if Luigi managed to fit a valid path which means **wrongPath** was only partially wrong. Otherwise it is false.
+  
+
+## Navigation properties
+The node navigation properties enable you to configure global navigation settings under the `navigation:` section in the configuration file.
+
+### nodeAccessibilityResolver
+- **type**: ? 
+- **required**: ?
```
Comment (zarkosimic) [inline docs/navigation-parameters-reference.md]: - **type**: any 
- **required**: no

File: docs/navigation-parameters-reference.md [hunk 5]
```diff
@@ -1,269 +1,293 @@
...
+- **description**: receives all values defined in the node configuration. It allows you to define a permission checker function that gets executed on every node. If it returns `false`, Luigi removes the node and its children from the navigation structure. See [angular navigation.js](../core/examples/luigi-sample-angular/src/luigi-config/extended/navigation.js) for an example.
+
+### defaults.isolateView
+- **type**: ? 
+- **required**: ?
```
Comment (zarkosimic) [inline docs/navigation-parameters-reference.md]: - **type**: boolean
- **required**: no

File: docs/navigation-parameters-reference.md [hunk 6]
```diff
@@ -1,269 +1,293 @@
...
+- **description**: renders all views in new frames. This setting overrides the same-domain frame reuse. The **defaults.isolateView** is disabled by default, and you can overwrite it using the **isolateView** value on a single node level.
+
+### preloadViewGroups
+- **type**: boolean
+- **required**: ?
+- **description**: allows deactivating the default preloading of [view groups](navigation-configuration.md#view-groups) iframes.
+
+### viewGroupsSettings
+- **type**: object
+- **required**: ?
```
Comment (zarkosimic) [inline docs/navigation-parameters-reference.md]: - **type**: object
- **required**: no

File: docs/navigation-parameters-reference.md [hunk 7]
```diff
@@ -1,269 +1,293 @@
...
+- **type**: object
+- **required**: no
+- **description**: indicates that the node links to an external URL. If this property is defined, **pathSegment** and **link** properties are ignored.
+- **attributes**:
+  - **sameWindow** defines if the external URL is opened in a new or current tab. The default value for this property is `false`.
+  - **url** is the external URL that the node leads to.
+
+### label
+- **type**: string
+- **required**: yes
```
Comment (zarkosimic) [inline docs/navigation-parameters-reference.md]: - **type**: string
- **required**: no

File: docs/navigation-parameters-reference.md [hunk 8]
```diff
@@ -1,269 +1,293 @@
...
+- **description**: contains the display name of the navigation node.
+
+### testId
+- **type**: string
+- **required**: no
+- **description**: allows you define your own custom `testId`. If there is nothing specified, it is a combination of the node's pathSegment followed by a dash (if pathSegment exists) and the label written as one word and lower case (e.g. `pathsegment_label` or `label`).
+
+### hideFromNav
+- **type**: ?
+- **required**: no
```
Comment (zarkosimic) [inline docs/navigation-parameters-reference.md]: - **type**: boolean
- **required**: no

File: docs/navigation-parameters-reference.md [hunk 9]
```diff
@@ -1,269 +1,293 @@
...
+* [Product switcher](#product-switcher)
+* [App switcher](#app-switcher)
+
+
+## Routing properties
+You can configure the way Luigi tackles routing in your application in the `routing:` section of the configuration file. For example, you can choose the routing strategy to apply in your application as either hash or path location routing.
+
+### useHashRouting
+- **type**: boolean 
+- **required**: yes
```
[the remark this answers is not in the document]
Comment (alexandra-simeonova) [reply docs/navigation-parameters-reference.md]: Just to clarify, is the default hash- or path-based routing? Perhaps this information would be useful to include (maybe in place of required since Philipp said nothing is required...)

File: docs/navigation-parameters-reference.md [hunk 10]
```diff
@@ -1,269 +1,293 @@
...
+- **description**: allows you define your own custom `testId`. If there is nothing specified, it is a combination of the node's pathSegment followed by a dash (if pathSegment exists) and the label written as one word and lower case (e.g. `pathsegment_label` or `label`).
+
+### hideFromNav
+- **type**: ?
+- **required**: no
+- **description**: shows or hides a navigation node. You can still navigate to the node but it does not show up in the top or left pane.
+
+### viewUrl
+- **type**: string
+- **required**: yes
```
Comment (zarkosimic) [inline docs/navigation-parameters-reference.md]: - **type**: string
- **required**: no

File: docs/navigation-parameters-reference.md [hunk 11]
```diff
@@ -1,294 +1,302 @@
...
+### category
+- **type**: string
+- **description**: defines a group of views separated with a headline and an icon. You should define at least one node in a group should as an Object with **label** and **icon** attributes. For all other nodes, you can set **category** as a string with the `label` value.
+- **attributes**:
+  - **label** is a string that represents the title of the category
+  - **icon** is the name of an icon, without the `sap-icon--` prefix. Its source may be [OpenUI](https://openui5.hana.ondemand.com/1.40.10/iconExplorer.html) or a custom link (relative or absolute) to an image. The icon is displayed next to the node label in the side navigation or instead of the label in the top navigation. In case you accidentally define different icons in a category group, only the first one is used.
+  - **collapsible** if set to `true`, category items are hidden at first. To expand them, click the main category node.
+  - **testId** is a string where you can define your own custom `testId`. If nothing is specified, it is the node's label written as one word and lower case (e.g. `label`).
+
+### opeNodeInModal
```
Comment (zarkosimic) [inline docs/navigation-parameters-reference.md]:
```suggestion
### openNodeInModal
```

File: docs/navigation-parameters-reference.md [hunk 12]
```diff
@@ -1,294 +1,302 @@
...
+- **type**: ?
```
Comment (zarkosimic) [inline docs/navigation-parameters-reference.md]:
```suggestion
- **type**: boolean or object
```

File: docs/navigation-parameters-reference.md [hunk 13]
```diff
@@ -1,294 +1,302 @@
...
+- **attributes**:
+  - **title** modal title. By default, it is the node label. If there is no label, it is left empty
+  - **size** **(`"l"` \| `"m"` \| `"s"`)** size of the modal (optional, default `"l"`)
+
+### onNodeActivation
+- **type**: function
+- **description**: executed when a request to navigate to the node occurs. As an input property, the function receives the node object as described in the configuration. This function can return results synchronously or asynchronously. If the function returns boolean `false`, the navigation is not triggered, otherwise, navigation renders as usual.
+
+### clientPermissions.changeCurrentLocale
+- **type**: ? boolean
```
Comment (zarkosimic) [inline docs/navigation-parameters-reference.md]:
```suggestion
- **type**: boolean
```

File: docs/navigation-parameters-reference.md [hunk 14]
```diff
@@ -1,294 +1,302 @@
...
+- **description**: current locale can be changed from client using the corresponding API if this is set to `true`
+- **example**: ?
+```javascript
+{
+  clientPermissions: {
+  changeCurrentLocale: true
+}
+```
```
Comment (zarkosimic) [inline docs/navigation-parameters-reference.md]: - **example**:
```javascript
  clientPermissions: {
      changeCurrentLocale: true
  }
```

File: docs/navigation-parameters-reference.md [hunk 15]
```diff
@@ -1,294 +1,302 @@
...
-- **defaultLabel** specifies the default label that is shown if no context is selected.
-- **parentNodePath** specifies the base path, that is prepended to **options[].pathValue**. It must be an absolute path.
-- **lazyloadOptions** defines when to fetch **options**. When set to `true`, loads **options** when you click the context switcher. It doesn't involve any caching. When set to `false`, loads **options** once the page loads. The default value is `true`. 
-- **options** defines the list of context element. Context element properties are:
+### defaultLabel
+- **type**: string
+- **description**: specifies the default label that is shown if no context is selected.
+
+### parentNodePath
+- **type**: ?
```
Comment (zarkosimic) [inline docs/navigation-parameters-reference.md]:
```suggestion
- **type**: string
```

File: docs/navigation-parameters-reference.md [hunk 16]
```diff
@@ -1,294 +1,302 @@
...
+- **description**: specifies the base path, that is prepended to **options[].pathValue**. It must be an absolute path.
+
+### lazyloadOptions
+- **type**: ? boolean
```
Comment (zarkosimic) [inline docs/navigation-parameters-reference.md]:
```suggestion
- **type**: boolean
```

File: docs/navigation-parameters-reference.md [hunk 17]
```diff
@@ -1,294 +1,302 @@
...
+- **description**: defines when to fetch **options**. When set to `true`, loads **options** when you click the context switcher. It doesn't involve any caching. When set to `false`, loads **options** once the page loads. The default value is `true`.
+
+### options
+- **type**: ?
```
Comment (zarkosimic) [inline docs/navigation-parameters-reference.md]:
```suggestion
- **type**: array
```

File: docs/navigation-parameters-reference.md [hunk 18]
```diff
@@ -1,294 +1,302 @@
...
+- **description**: defines the list of context element.
+- **attributes**:
   - **label** defines the context element label. If not defined, the **pathValue** is passed to **fallbackLabelResolver** to set its value. The default value is **pathValue**, if **fallbackLabelResolver** is not defined.
   - **pathValue** defines the context element path that is appended to **parentNodePath** and reflects a **pathSegment**.
-- **actions** defines a list of additional elements that are shown on above or below the context switcher **options**. Each action contains the following parameters:
+
+### actions
+- **type**: ?
```
Comment (zarkosimic) [inline docs/navigation-parameters-reference.md]:
```suggestion
- **type**: array
```

File: docs/navigation-parameters-reference.md [hunk 19]
```diff
@@ -1,294 +1,302 @@
...
-  - **position** defines the action element position. Can be `top` or `bottom`. The default value is `top`. This parameter is optional.
-  - **link** defines an absolute Link to a **node**. This parameter is optional.
+  - **position** defines the action element position. Can be `top` or `bottom`. The default value is `top`. This property is optional.
+  - **link** defines an absolute Link to a **node**. This property is optional.
   - **clickHandler** specifies a function and is executed on click and should return a boolean. If it returns `true`, **link** is opened afterwards.
-- **fallbackLabelResolver** specifies a function used to fetch the **label** for **options** with no **label** defined. Additionally, it fetches the drop-down label for non-existing **options**.
-- **preserveSubPathOnSwitch** if set to `true`, the sub-path will be preserved on context switch.
+
+### fallbackLabelResolver
+- **type**: ?
```
Comment (zarkosimic) [inline docs/navigation-parameters-reference.md]:
```suggestion
- **type**: any
```

File: docs/navigation-parameters-reference.md [hunk 20]
```diff
@@ -1,294 +1,302 @@
...
+- **description**: specifies a function used to fetch the **label** for **options** with no **label** defined. Additionally, it fetches the drop-down label for non-existing **options**.
+
+### preserveSubPathOnSwitch
+- **type**: ?
```
Comment (zarkosimic) [inline docs/navigation-parameters-reference.md]:
```suggestion
- **type**: boolean
```

File: docs/navigation-parameters-reference.md [hunk 21]
```diff
@@ -1,294 +1,302 @@
...
+- **description**: if set to `true`, the sub-path will be preserved on context switch.
 
 
 ## Profile
 
 The profile section is a configurable drop-down list available in the top navigation bar. Within the configuration, you can override the logout item content (if authorization is configured) and/or add links to Luigi navigation nodes. To do so, add the **profile** property to the **navigation** object using the following optional properties:
 
-- **logout** overrides the content of the logout item.
+### logout
+- **type**: ?
```
Comment (zarkosimic) [inline docs/navigation-parameters-reference.md]:
```suggestion
- **type**: object
```

File: docs/navigation-parameters-reference.md [hunk 22]
```diff
@@ -1,294 +1,302 @@
...
   - **icon** overrides the icon for the logout item. The default value is [SAP UI5 log icon](https://sapui5.hana.ondemand.com/test-resources/sap/m/demokit/iconExplorer/webapp/index.html#/overview/SAP-icons/?tag=logout).
-  - **customLogoutFn** defines a function to implement your own logout functionality. It is recommended to only use this function if no IDP is configured. If an IDP with a corresponding [logout function](https://github.com/SAP/luigi/blob/master/docs/authorization-configuration.md) is defined , the customLogoutFn on profile will be ignored.
-- **items** is an array of objects, each one being a link to a Luigi navigation node or an external URL. An item can have the following parameters:
+  - **customLogoutFn** defines a function to implement your own logout functionality. It is recommended to only use this function if no IDP is configured. If an IDP with a corresponding [logout function](authorization-configuration.md) is defined , the customLogoutFn on profile will be ignored.
+
+### items
+- **type**: ?
```
Comment (zarkosimic) [inline docs/navigation-parameters-reference.md]:
```suggestion
- **type**: array
```

File: docs/navigation-parameters-reference.md [hunk 23]
```diff
@@ -1,294 +1,302 @@
...
-  - **externalLink** is an object which indicates that the node links to an external URL. If this parameter is defined, the **link** parameter is ignored. It has the following properties:
-    - **sameWindow** defines if the external URL is opened in the current tab or in a new one. The default value for this parameter is `false`.
+  - **externalLink** is an object which indicates that the node links to an external URL. If this property is defined, the **link** property is ignored. It has the following attributes:
+    - **sameWindow** defines if the external URL is opened in the current tab or in a new one. The default value for this property is `false`.
     - **url** is the external URL that the link leads to.
-- **staticUserInfoFn** is a function that is used to retrieve a user's name and email to simulate a login. It can be used when authorization is disabled and also gets called if the defined IDP provider does not have `settings.userInfoFn` defined or does not provide a `userInfo` function internally. It can be asynchronous and should return an object with `name`, `email` and `picture` properties.
+
+### staticUserInfoFn
+- **type**: ? function
```
Comment (zarkosimic) [inline docs/navigation-parameters-reference.md]:
```suggestion
- **type**: function
```

File: docs/navigation-parameters-reference.md [hunk 24]
```diff
@@ -1,294 +1,302 @@
...
+### label
+- **type**: string
+- **description**: defines the label of the product switcher. It is displayed as a title attribute on hover in the top navigation and as a headline in the mobile pop-up.
+
+### testId
+- **type**: string
+- **description**: enables you to define your own custom `testId`. If nothing is specified, it is the node's label written as one word and lower case (e.g. `label`).
+
+### icon
+- **type**: string ?
```
Comment (zarkosimic) [inline docs/navigation-parameters-reference.md]:
```suggestion
- **type**: string
```

File: docs/navigation-parameters-reference.md [hunk 25]
```diff
@@ -1,294 +1,302 @@
...
-- **showMainAppEntry** includes the link to the root of the Luigi application in the drop-down using the **title** specified in the **settings/header** section of the configuration as a label.
-- **items** defines the list of apps. App element properties are:
-  - **title** defines the app title. This is shown in the app switcher drop-down as well as the title in the header of the Luigi application if a user is in the context of the app. 
-  - **subTitle** defines the app sub-title. This is shown as the sub-title in the header of the Luigi application if a user is in the context of the app. 
-  - **link** is a link within the Luigi application that defines the root of the app. It is used to switch to the app if the drop-down entry is selected. It is also used to determine if a user is within the app's scope, so that the corresponding title and sub-title can be rendered in the header. 
+
+### showMainAppEntry
+- **type**: array
```
Comment (zarkosimic) [inline docs/navigation-parameters-reference.md]:
```suggestion
- **type**: boolean
```

File: docs/navigation-parameters-reference.md [hunk 26]
```diff
@@ -1,294 +1,302 @@
...
+- **description**: includes the link to the root of the Luigi application in the drop-down using the **title** specified in the **settings/header** section of the configuration as a label.
```
Comment (zarkosimic) [inline docs/navigation-parameters-reference.md]: I see this only as a boolean type. Might we would need to check it again.

File: docs/navigation-parameters-reference.md [hunk 27]
```diff
@@ -1,294 +1,302 @@
...
+
+### items
+- **type**: ?
```
Comment (zarkosimic) [inline docs/navigation-parameters-reference.md]:
```suggestion
- **type**: array
```

File: docs/navigation-parameters-reference.md [hunk 28]
```diff
@@ -1,294 +1,305 @@
...
+
+### preloadViewGroups
+- **type**: boolean
+- **description**: allows deactivating the default preloading of [view groups](navigation-configuration.md#view-groups) iframes.
+
+### viewGroupsSettings
+- **type**: object
+- **description**: contains key-object pairs, where the key is the view group name as specified in the node properties, and the object contains key-value pairs. In each key-value pair, the key is the feature name and the value is the actual setting. The following options are supported:
+- **attributes**:
+  - **preloadUrl**(string): needs to be an absolute URL for a node from the view group. It is recommended that you use a dedicated small, visually empty view, which imports Luigi Client and is fine with getting an empty context, for example, without an access token. The **preloadUrl** property is also required for view group caching in case you need a view group iframe to refresh whenever you navigate back to it.
```
Comment (pekura) [inline docs/navigation-parameters-reference.md]: "needs to be an absolute URL for a node from the view group" - I would rephrase it as the preload URL may not be an URL associated with any node. It is a view URL of a view belonging to the view group. It is a URL on the microfrontend side.

File: docs/navigation-parameters-reference.md [hunk 29]
```diff
@@ -1,294 +1,305 @@
...
+## Routing properties
+You can configure the way Luigi tackles routing in your application in the `routing:` section of the configuration file. For example, you can choose the routing strategy to apply in your application as either hash or path location routing.
+
+### useHashRouting
+- **type**: boolean
+- **description**: defines either hash-based (`url.com/#/yourpath`) or path-based (`url.com/yourpath`) routing.
+
+### nodeParamPrefix
+- **type**: string
+- **description**: sets the prefix character when using the `LuigiClient.linkManager().withParam()` function, which provides a way to simply attach query properties to the view URL for activities such as sorting and filtering. The URL contains the properties to allow deep linking. If you want to use a different character prefix, define yours here. The default character is `~`.
```
Comment (pekura) [inline docs/navigation-parameters-reference.md]: We should explain that the prefix here is necessary to avoid potential conflicts with query parameters belonging to the Luigi app itself. Only query parameters with that prefix are passed to microfrontends.

File: docs/navigation-parameters-reference.md [hunk 30]
```diff
@@ -1,294 +1,305 @@
...
+- **type**: string
+- **description**: allows you define your own custom `testId`. If there is nothing specified, it is a combination of the node's pathSegment followed by a dash (if pathSegment exists) and the label written as one word and lower case (e.g. `pathsegment_label` or `label`).
+
+### hideFromNav
+- **type**: boolean
+- **description**: shows or hides a navigation node. You can still navigate to the node but it does not show up in the top or left pane.
+
+### viewUrl
+- **type**: string
+- **description**: contains the URL or path to a view which renders when you click the navigation node. Use either a full URL or a relative path. This value may consist of variables if you have specified a **navigationContext** with a dynamic **pathSegment**. If **viewUrl** is undefined, Luigi activates the child node specified in **defaultChildNode**. When both **viewUrl** and **defaultChildNode** are undefined, Luigi opens the first child of the current node.
```
Comment (pekura) [inline docs/navigation-parameters-reference.md]: Hm, the viewUrl can contain variables from:
- dynamic path segment (with ':' prefix, e.g. :productId)
- node params, in form {nodeParams.paramName}
- context, in form {context.contextPropertyName}

It has nothing to do with the navigationContext.

File: docs/navigation-parameters-reference.md [hunk 31]
```diff
@@ -1,294 +1,305 @@
...
-          testId: 'myTestId',
-          link: '/projects/pr1'
-        }
-      ]
-    },
-  }
-});
+### loadingIndicator.hideAutomatically
+- **type**: boolean
+- **description**: disables the automatic hiding of the loading indicator once the micro frontend is loaded. It is only considered if the loading indicator is enabled. It does not apply if the loading indicator is activated manually with the `LuigiClient.uxManager().showLoadingIndicator()` function. If the loading indicator is enabled and automatic hiding is disabled, use `LuigiClient.uxManager().hideLoadingIndicator()` to hide it manually in your micro frontend during the startup. This property is enabled by default.
```
Comment (pekura) [inline docs/navigation-parameters-reference.md]: "disables the automatic hiding of the loading indicator..." if set to false!

File: docs/navigation-parameters-reference.md [hunk 32]
```diff
@@ -1,294 +1,305 @@
...
+- **attributes**:
+  - **label** is a string that represents the title of the category
+  - **icon** is the name of an icon, without the `sap-icon--` prefix. Its source may be [OpenUI](https://openui5.hana.ondemand.com/1.40.10/iconExplorer.html) or a custom link (relative or absolute) to an image. The icon is displayed next to the node label in the side navigation or instead of the label in the top navigation. In case you accidentally define different icons in a category group, only the first one is used.
+  - **collapsible** if set to `true`, category items are hidden at first. To expand them, click the main category node.
+  - **testId** is a string where you can define your own custom `testId`. If nothing is specified, it is the node's label written as one word and lower case (e.g. `label`).
+
+### openNodeInModal
+- **type**: boolean or object
+- **description**: adds a badge with a number and a label to a node. Nodes that are part of a category show a cumulated number of all badges in this category. **badgeCounter** is only available for top navigation items.
```
Comment (pekura) [inline docs/navigation-parameters-reference.md]: The description does not match the functionality.

File: docs/navigation-parameters-reference.md (hunk 26 above)
Comment (alexandra-simeonova) [reply docs/navigation-parameters-reference.md]: According to Pete: "if it is set to true then in the app switcher dropdown an entry is shown that points to the app root, and the title of the entry is the title of the app (that is why we only need a boolean here)"

File: docs/navigation-parameters-reference.md [hunk 33]
```diff
@@ -1,295 +1,309 @@
...
+  - **sameWindow** defines if the external URL is opened in a new or current tab. The default value for this property is `false`.
+  - **url** is the external URL that the node leads to.
+
+### label
+- **type**: string
+- **description**: contains the display name of the navigation node.
+
+### testId
+- **type**: string
+- **description**: allows you define your own custom `testId`. If there is nothing specified, it is a combination of the node's pathSegment followed by a dash (if pathSegment exists) and the label written as one word and lower case (e.g. `pathsegment_label` or `label`).
```
Comment (bszwarc) [inline docs/navigation-parameters-reference.md]:
```suggestion
- **description**: allows you to define your own custom **testId**. If you do not specify it, it is a combination of the node's **pathSegment** followed by an underscore and the label, written as one word in lower case. If the **pathSegment** does not exist, the **testId** includes the label only. This way, you can have `pathsegment_label` or `label`.
```

File: docs/navigation-parameters-reference.md [hunk 34]
```diff
@@ -1,295 +1,309 @@
...
-- **onNodeActivation** is an optional function executed when a request to navigate to the node occurs. As an input parameter, the function receives the node object as described in the configuration. This function can return results synchronously or asynchronously. If the function returns boolean `false`, the navigation is not triggered, otherwise, navigation renders as usual.
-- **clientPermissions.changeCurrentLocale** current locale can be changed from client using the corresponding API if this is set to `true`
-- **tabNav** is a boolean which renders the children of the node as a horizontal navigation bar. Sub-children are not supported. When you categorize nodes you will get a drop-down menu in the horizontal navigation.
+
+### category
+- **type**: string or object
+- **description**: defines a group of views separated with a headline and an icon. You should define at least one node in a group should as an Object with **label** and **icon** attributes. For all other nodes, you can set **category** as a string with the `label` value.
+- **attributes**:
+  - **label** is a string that represents the title of the category
```
Comment (bszwarc) [inline docs/navigation-parameters-reference.md]:
```suggestion
  - **label** is a string that represents the title of the category.
```

File: docs/navigation-parameters-reference.md [hunk 35]
```diff
@@ -1,295 +1,309 @@
...
+  - **icon** is the name of an icon, without the `sap-icon--` prefix. Its source may be [OpenUI](https://openui5.hana.ondemand.com/1.40.10/iconExplorer.html) or a custom link (relative or absolute) to an image. The icon is displayed next to the node label in the side navigation or instead of the label in the top navigation. In case you accidentally define different icons in a category group, only the first one is used.
+  - **collapsible** if set to `true`, category items are hidden at first. To expand them, click the main category node.
+  - **testId** is a string where you can define your own custom `testId`. If nothing is specified, it is the node's label written as one word and lower case (e.g. `label`).
```
Comment (bszwarc) [inline docs/navigation-parameters-reference.md]:
```suggestion
  - **testId** is a string where you can define your own custom `testId`. If nothing is specified, it is the node's label written as one word in lower case, for example`label`.
```

File: docs/navigation-parameters-reference.md [hunk 36]
```diff
@@ -1,295 +1,309 @@
...
+
+### openNodeInModal
+- **type**: boolean or object
+- **description**:  configures the settings of a view which opens in a modal. You can set the openNodeInModal property to true to use the default modal title and size, or you can specify them using these attributes:
```
Comment (bszwarc) [inline docs/navigation-parameters-reference.md]:
```suggestion
- **description**:  configures the settings of a view which opens in a modal. You can set the **openNodeInModal** property to `true` to use the default modal title and size, or you can specify them using these attributes:
```

File: docs/navigation-parameters-reference.md [hunk 37]
```diff
@@ -1,295 +1,309 @@
...
+- **attributes**:
+  - **title** is the modal title. By default, it is the node label. If there is no label, it is left empty.
+  - **size** **(`"l"` \| `"m"` \| `"s"`)** specifies the size of the modal. The default size is **`"l"`**, which means 80%. You can also use **`"m"`** (60%) and **`"s"`** (40%) to set the modal size.
```
Comment (bszwarc) [inline docs/navigation-parameters-reference.md]: Are the values here escaped on purpose? If not, you can change the formatting to regular one. ** for property, ` for value.

File: docs/navigation-parameters-reference.md [hunk 38]
```diff
@@ -1,295 +1,309 @@
...
+- **example**:
+```javascript
+  clientPermissions: {
+    changeCurrentLocale: true
+  }
+```
+
+### tabNav
+- **type**: boolean
+- **description**: tabNav renders the children of the node as a horizontal navigation bar. Sub-children are not supported. When you categorize nodes you will get a drop-down menu in the horizontal navigation.
```
Comment (bszwarc) [inline docs/navigation-parameters-reference.md]:
```suggestion
- **description**: renders the children of the node as a horizontal navigation bar. Sub-children are not supported. When you categorize nodes you will get a drop-down menu in the horizontal navigation.
```

File: docs/navigation-parameters-reference.md [hunk 39]
```diff
@@ -1,295 +1,309 @@
...
-- **fallbackLabelResolver** specifies a function used to fetch the **label** for **options** with no **label** defined. Additionally, it fetches the drop-down label for non-existing **options**.
-- **preserveSubPathOnSwitch** if set to `true`, the sub-path will be preserved on context switch.
+
+### fallbackLabelResolver
+- **type**: any
+- **description**: specifies a function used to fetch the **label** for **options** with no **label** defined. Additionally, it fetches the drop-down label for non-existing **options**.
+
+### preserveSubPathOnSwitch
+- **type**: boolean
+- **description**: if set to `true`, the sub-path will be preserved on context switch.
```
Comment (bszwarc) [inline docs/navigation-parameters-reference.md]:
```suggestion
- **description**: if set to `true`, the sub-path is preserved on context switch.
```

File: docs/navigation-parameters-reference.md [hunk 40]
```diff
@@ -1,295 +1,309 @@
...
   - **testId** is a string where you can define your own custom `testId`. If nothing is specified, it is the node's label written as one word and lower case (e.g. `label`).
   - **icon** overrides the icon for the logout item. The default value is [SAP UI5 log icon](https://sapui5.hana.ondemand.com/test-resources/sap/m/demokit/iconExplorer/webapp/index.html#/overview/SAP-icons/?tag=logout).
-  - **customLogoutFn** defines a function to implement your own logout functionality. It is recommended to only use this function if no IDP is configured. If an IDP with a corresponding [logout function](https://github.com/SAP/luigi/blob/master/docs/authorization-configuration.md) is defined , the customLogoutFn on profile will be ignored.
-- **items** is an array of objects, each one being a link to a Luigi navigation node or an external URL. An item can have the following parameters:
+  - **customLogoutFn** defines a function to implement your own logout functionality. It is recommended to only use this function if no IDP is configured. If an IDP with a corresponding [logout function](authorization-configuration.md) is defined , the customLogoutFn on profile will be ignored.
```
Comment (bszwarc) [inline docs/navigation-parameters-reference.md]:
```suggestion
  - **customLogoutFn** defines a function to implement your own logout functionality. Use this function only if no IDP is configured. If you define IDP with a corresponding [logout function](authorization-configuration.md), the **customLogoutFn** set for a profile is ignored.
```

File: docs/navigation-parameters-reference.md [hunk 41]
```diff
@@ -1,295 +1,309 @@
...
+    - **sameWindow** defines if the external URL is opened in the current tab or in a new one. The default value for this property is `false`.
     - **url** is the external URL that the link leads to.
-- **staticUserInfoFn** is a function that is used to retrieve a user's name and email to simulate a login. It can be used when authorization is disabled and also gets called if the defined IDP provider does not have `settings.userInfoFn` defined or does not provide a `userInfo` function internally. It can be asynchronous and should return an object with `name`, `email` and `picture` properties.
+
+### staticUserInfoFn
+- **type**: function
+- **description**: a function that is used to retrieve a user's name and email to simulate a login. It can be used when authorization is disabled and also gets called if the defined IDP provider does not have `settings.userInfoFn` defined or does not provide a `userInfo` function internally. It can be asynchronous and should return an object with `name`, `email` and `picture` properties.
```
Comment (bszwarc) [inline docs/navigation-parameters-reference.md]:
```suggestion
- **description**: used to retrieve a user's name and email to simulate logging in. It can be used when authorization is disabled and also gets called if the defined IDP provider does not have **settings.userInfoFn** defined or does not provide a `userInfo` function internally. It can be asynchronous and should return an object with **name**, **email** and **picture** properties.
```

File: docs/navigation-parameters-reference.md [hunk 42]
```diff
@@ -1,295 +1,309 @@
...
-  - **subTitle** defines the app sub-title. This is shown as the sub-title in the header of the Luigi application if a user is in the context of the app. 
-  - **link** is a link within the Luigi application that defines the root of the app. It is used to switch to the app if the drop-down entry is selected. It is also used to determine if a user is within the app's scope, so that the corresponding title and sub-title can be rendered in the header. 
+
+### showMainAppEntry
+- **type**: boolean
+- **description**: includes the link to the root of the Luigi application in the drop-down using the **title** specified in the **settings/header** section of the configuration as a label.
+
+### items
+- **type**: array
+- **description**: defines the list of apps. App element attributes are described below.
```
Comment (bszwarc) [inline docs/navigation-parameters-reference.md]: Apps or applications? I'm fine with either one of them, just asking :)

Comment (alexandra-simeonova) [reply docs/navigation-parameters-reference.md]: In the name **appSwitcher**, app stands for application. So applications is better, I think :)
